src/test/java/edu/ucsb/cs156/example/controllers/RecommendationControllerTests.java

```diff
@@ -0,0 +1,377 @@
+                                .professorEmail("teacher@ucsb.edu")
+                                .explanation("bad lectures")
+                                .dateRequested(dR1)
+                                .dateNeeded(dN1)
+                                .done(false)
+                                .build();
+                
+                LocalDateTime dR2 = LocalDateTime.parse("2022-01-04T00:00:00");
+                LocalDateTime dN2 = LocalDateTime.parse("2022-02-05T00:00:00");
+
+                Recommendation recommendation2 = Recommendation.builder()
+                                .requesterEmail("johnjoe@ucsb.edu")
+                                .professorEmail("prof@ucsb.edu")
+                                .explanation("more homework")
+                                .dateRequested(dR2)
+                                .dateNeeded(dN2)
+                                .done(false)
+                                .build();
+
+                ArrayList<Recommendation> expectedRecs = new ArrayList<>();
+                expectedRecs.addAll(Arrays.asList(recommendation1, recommendation2));
+
+                when(recommendationRepository.findAll()).thenReturn(expectedRecs);
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/Recommendation/all"))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+
+                verify(recommendationRepository, times(1)).findAll();
+                String expectedJson = mapper.writeValueAsString(expectedRecs);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void an_admin_user_can_post_a_new_recommendation() throws Exception {
+                // arrange
+
+                LocalDateTime dR = LocalDateTime.parse("2022-01-03T00:00:00");
+                LocalDateTime dN = LocalDateTime.parse("2022-02-03T00:00:00");
+
+                Recommendation recommendation = Recommendation.builder()
+                                .requesterEmail("jgaucho@ucsb.edu")
+                                .professorEmail("teacher@ucsb.edu")
+                                .explanation("badLectures")
+                                .dateRequested(dR)
+                                .dateNeeded(dN)
+                                .done(false)
+                                .build();
+
+                when(recommendationRepository.save(eq(recommendation))).thenReturn(recommendation);
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                post("/api/Recommendation/post?requesterEmail=jgaucho@ucsb.edu&professorEmail=teacher@ucsb.edu&explanation=badLectures&dateRequested=2022-01-03T00:00:00&dateNeeded=2022-02-03T00:00:00")
```

Comment:
You'll need to add the `done` parameter here.

Reply:
fixed in most recent commit